### PR TITLE
Add Python zqd API client

### LINKED ISSUE #1519: python zqd api client
As a first step towards experimenting with Jupyter notebooks, we need a Python module that can be used to issue ZQL queries against a zqd endpoint, obtaining responses in zjson and translating to native Python objects.

### PR DESCRIPTION
**Pull request by nwt**
Install this branch with `pip install 'git+https://github.com/brimsec/zq@python-zqd#subdirectory=python/zqd'`.

After merge, install master with `pip install 'git+https://github.com/brimsec/zq#subdirectory=python/zqd'`.

Closes #1519.

JupyterLab action shot:
image

### CODE REVIEW

**Comment by mattnibs** [comment]
Can we get a test with this? I know it would be a pain, having said that it would be nice to know if an api change broke this.

**Comment by nwt** [comment]
@mattnibs: Added a test in c47bdb9c4fca5f3f065b9753748197c2e9c9e068.

**Comment by mccanne** [comment]
very slick!

File: python/zqd/zqd.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        r = self.session.post(self.base_url + "/ast", json={'zql': zql})
+        r.raise_for_status()
+        return r.json()
+
+    def search(self, space_name, zql):
+        aliases, schemas = {}, {}
+        for raw in self.search_raw(space_name, zql):
+            if raw['type'] == 'SearchRecords':
+                for rec in raw['records']:
+                    if 'aliases' in rec:
```
Comment (mccanne) [inline python/zqd/zqd.py:27]: It seems like the the process of parsing zqd search results into a stream of zjson JSON objects should be separate from the thing that parses a stream of zjson JSON objects into a stream of native python objects.  This way, for example you could have a python program that reads a file that has zjson in and parses the zjson into the python objects.  This should also help people reading the code to understand the model here more easily.

Comment (mccanne) [reply python/zqd/zqd.py:27]: This would also make tests easier to write as this subsystem evolves.

Comment (nwt) [reply python/zqd/zqd.py:27]: @mccanne: Is this what you had in mind?

```python
class Client():
    ...
    def search(self, space_name, zql):
        raw = self.search_raw(space_name, zql)
        zjson = take_zjson(raw)
        return decode_zjson(zjson)


def take_zjson(raw):
    for obj in raw:
        if obj['type'] == 'SearchRecords':
            for zjson in obj['records']:
                yield zjson


def decode_zjson(zjson):
    aliases, schemas = {}, {}
    for obj in zjson:
        if 'aliases' in obj:
            for a in obj['aliases']:
                aliases[a['name']] = a['type']
        if 'schema' in obj:
            schemas[obj['id']] = obj['schema']
        yield _to_native(aliases, schemas[obj['id']], obj['values'])
```

**Review by mattnibs — APPROVED** [review]
lgtm

File: python/zqd/zqd.py (hunk 1 above)
Comment (mccanne) [reply python/zqd/zqd.py:27]: I think so.

Comment (nwt) [reply python/zqd/zqd.py]: Done in cf76c327ba99e196236e177c0f9a259c7e64c135.

**Review by alfred-landrum — APPROVED** [review]
I mentioned to noah on slack that we could add a note to the contributing file that we expect python3 to be available in the dev environment.

**Pull request merged**